custom/inddex/filters.py

```diff
@@ -0,0 +1,188 @@
+        ]
+
+
+class PregnancyFilter(BaseSingleOptionFilter):
+    slug = 'pregnant'
+    label = _('Pregnancy')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('yes', _('Yes')),
+            ('no', _('No')),
+            ('refuse_to_answer', _('Refuse to answer')),
+            ('dont_know', _('Don\'t know'))
+        ]
+
+
+class SettlementAreaFilter(BaseSingleOptionFilter):
+    slug = 'urban_rural'
+    label = _('Urban/Rural')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('urban', _('Urban')),
+            ('rural', _('Rural'))
+        ]
+
+
+class BreastFeedingFilter(BaseSingleOptionFilter):
+    slug = 'breastfeeding'
+    label = _('Breastfeeding')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        # TODO: Check value in production application ?? refuse_de_rpondre
+        return [
+            ('yes', _('Yes')),
+            ('no', _('No')),
+            ('refuse_de_rpondre', _('Refuse to answer'))
+        ]
+
+
+class SupplementsFilter(BaseSingleOptionFilter):
+    slug = 'supplements'
+    label = _('Supplement Use')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('yes', _('Yes')),
+            ('no', _('No'))
+        ]
+
+
+class RecallStatusFilter(BaseSingleOptionFilter):
+    slug = 'recall_status'
+    label = _('Recall Status')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('Open', _('Not Completed')),
+            ('Completed', _('Completed'))
+        ]
+
+
+class GapDescriptionFilter(BaseSingleOptionFilter):
+    slug = 'gap_description'
+    label = _('Gap description')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('conv factor available', 'conv factor available'),
+            ('fct data available', 'fct data available'),
+            ('no conversion factor available', 'no conversion factor available'),
+            ('no fct data available', 'no fct data available'),
+            ('not applicable', 'not applicable'),
+            (
+                'using conversion factor from the reference food code',
+                'using conversion factor from the reference food code'
+            ),
+            ('using fct data from the reference food code', 'using fct data from the reference food code'),
+        ]
+
+
+class GapTypeFilter(BaseSingleOptionFilter):
+    slug = 'gap_type'
+    label = _('Gap type')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('fct', 'fct'),
+            ('conversion factor', 'conversion factor'),
+        ]
+
+
+class FoodCodeFilter(BaseSingleOptionFilter):
+    slug = 'food_code'
+    label = _('Food Code')
+    default_text = _('All')
+
+    @property
+    def food_codes(self):
+        return FoodCodeData(config={'domain': self.domain}).rows
```

Comment:
I've implemented this filter this way, because similar filters for IntraHealth dashboards were implemented like that